pyfifotax/report_data.py

```diff
@@ -0,0 +1,535 @@
+                tmp = self.held_shares[event.symbol].pop(
+                    event.quantity, event.sell_price, event.date
+                )
+                self.sold_shares[event.symbol].extend(tmp)
+
+                self.held_forex[event.currency].push(event.received_forex)
+
+                if event.paid_fees is not None:
+                    self.misc["Fees"].append(event.paid_fees)
+
+                self.awv_z10_events.append(
+                    awv.AWVEntryZ10Sale(
+                        date=event.date,
+                        symbol=event.symbol,
+                        currency=event.currency,
+                        quantity=event.quantity,
+                        value=event.quantity * event.sell_price,
+                    )
+                )
+
+            elif isinstance(event, MoneyDepositEvent):
+                new_forex = FIFOForex(
+                    currency=event.currency,
+                    quantity=(
+                        event.amount - event.fees.amount
+                        if event.fees is not None
+                        else event.amount
+                    ),
+                    buy_date=event.buy_date,
+                    source=f"Deposit of Money",
+                )
+                self.held_forex[event.currency].push(new_forex)
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+
+            elif isinstance(event, MoneyWithdrawalEvent):
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+                try:
+                    tmp = self.held_forex[event.currency].pop(
+                        event.amount,
+                        to_decimal(1),
+                        event.date,
+                    )
+                except ValueError:
+                    raise ValueError(
+                        f"Cannot withdraw {event.amount} {event.currency}, not enough owned."
+                    )
+                self.withdrawn_forex[event.currency].extend(tmp)
+
+            elif isinstance(event, CurrencyConversionEventFromEURToForex):
+                # "buy" forex (note: EUR used up not tracked)
+                new_forex = FIFOForex(
+                    currency=event.target_currency,
+                    quantity=event.foreign_amount,
+                    buy_date=event.date,
+                    source=f"Currency Conversion EUR to {event.target_currency}",
+                )
+                self.held_forex[event.target_currency].push(new_forex)
+                if event.source_fees is not None:
+                    self.misc["Fees"].append(event.source_fees)
+
+            elif isinstance(event, CurrencyConversionEventFromForexToEUR):
+                # "sell" forex (note: EUR received not tracked)
+                tmp = self.held_forex[event.source_currency].pop(
+                    event.foreign_amount,
```

Comment:
It's very much possible that again I have a misunderstanding how this works, but I found these in the logs:

```
[INFO] Converted money on 2019-04-10: from USD to 1.66 EUR
[DEBUG] [Σ 2.29 USD] Popping 0.14 USD (as a part of 1.66 USD) from the queue which was acquired on 2019-04-10
[DEBUG] [Σ 2.02 USD] Popping 0.27 USD (as a part of 1.52 USD) from the queue which was acquired on 2019-04-10
[DEBUG] [Σ 1.88 USD] Popping 0.14 USD (as a part of 1.25 USD) from the queue which was acquired on 2019-04-10
[DEBUG] [Σ 0.77 USD] Popping 1.11 USD from the queue which was acquired on 2019-04-10

[INFO] Converted money on 2019-06-03: from USD to 4.34 EUR
[DEBUG] [Σ 4.89 USD] Popping 0.77 USD (as a part of 4.34 USD) from the queue which was acquired on 2019-06-03
[DEBUG] [Σ 1.32 USD] Popping 3.57 USD from the queue which was acquired on 2019-06-03

[INFO] Converted money on 2019-06-14: from HUF to 63.12 EUR
[DEBUG] [Σ 305.83 HUF] Popping 63.12 HUF from the queue which was acquired on 2019-06-14
```

In all these cases it seems that the `source_currency` queue would be reduced by the amount of `foreign_amount`.